src/utils/websocket.js

```diff
@@ -13,11 +15,15 @@ function constructWebsocketURL(type, period) {
 
 export function wsChanges(dispatch, type, period) {
   let url = constructWebsocketURL(type, period)
-  const ws = new WebSocket(url)
-  ws.onmessage = function (msg) {
+  const options = {
+    connectionTimeout: 10000,
```

Comment:
The default of `4000` is more responsive. I've heard the figure `500ms` quoted as a reasonable inter-continental RTT. There could be other high latency situations that I'm not considering.